front/pages/api/w/[wId]/members/[userId]/index.ts

```diff
@@ -79,28 +79,73 @@ async function handler(
 
   switch (req.method) {
     case "POST":
-      if (
-        !req.body ||
-        !req.body.role ||
-        !["admin", "builder", "user", "revoked"].includes(req.body.role)
-      ) {
-        return apiError(req, res, {
-          status_code: 400,
-          api_error: {
-            type: "invalid_request_error",
-            message: "The request body is invalid, expects { role: string }.",
-          },
+      // TODO(@fontanierh): use DELETE for revoking membership
+      if (req.body.role === "revoked") {
+        const revokeResult = await MembershipResource.revokeMembership({
+          userId,
+          workspace: owner,
         });
-      }
+        if (revokeResult.isErr()) {
+          switch (revokeResult.error.type) {
+            case "not_found":
+              return apiError(req, res, {
+                status_code: 404,
+                api_error: {
+                  type: "workspace_user_not_found",
+                  message: "Could not find the membership.",
+                },
+              });
+            case "already_revoked":
+              // Should not happen, but we ignore.
+              break;
+            default:
+              assertNever(revokeResult.error.type);
+          }
+        }
 
-      await membership.update({
-        role: req.body.role,
-        endAt: req.body.role === "revoked" ? new Date() : null,
-      });
-      if (req.body.role === "revoked") {
         await updateWorkspacePerSeatSubscriptionUsage({
           workspaceId: owner.sId,
         });
+      } else {
+        const role = req.body.role;
```

Comment:
I am changing what is strictly needed for this refactoring. I am trying to avoid sneaking other unrelated refactors into this PR. I am happy to move this endpoint to io-ts separately though.